### PR TITLE
Feature/mob jump

### PR DESCRIPTION
**Pull request by lssalgado**
* Creating Entity scene with default movement logic;
* Improving mob AI to be able to Jump.

### CODE REVIEW

File: scripts/Mob.gd [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
+	$Ledge.position.x *= -1
+	$WallCollision.position.x *= -1
+	$WallCollision2.position.x *= -1
+
+func process_horizontal_movement(delta: float):
+	.process_horizontal_movement(delta)
+
+	if is_on_floor():
+		if $WallCollision.is_colliding() and $WallCollision2.is_colliding() == false:
+			.jump()
```
Comment (lssalgado) [inline scripts/Mob.gd]:
```suggestion
			jump()
```

We normally call the super method from the same overridden method. Here we can simply call `jump()`.

File: scripts/Entity.gd [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+var motion: Vector2 = Vector2.ZERO
+
+func process_horizontal_movement(delta: float):
+	motion.x += direction * ACCELERATION * delta
+	motion.x = clamp(motion.x, -max_speed, max_speed)
+
+func process_vertical_movement(delta: float):
+	motion.y += GRAVITY * delta
+
+func jump():
```
Comment (lssalgado) [inline scripts/Entity.gd:25]: I think that we could improve this method so `Player` could call it too.

File: scripts/Mob.gd [hunk 3]
```diff
@@ -1,38 +1,28 @@
...
-		$Ledge.position.x *= -1
-	
-func process_vertical_movement(delta):
-	motion.y += GRAVITY * delta
-	
+func change_direction():
+	direction = direction * -1
+	$Ledge.position.x *= -1
+	$WallCollision.position.x *= -1
+	$WallCollision2.position.x *= -1
```
Comment (lssalgado) [inline scripts/Mob.gd:12]: `WallCollision2` is not a good name. It should be more self-explanatory.

File: scripts/Entity.gd (hunk 2 above)
Comment (uSirThunder) [reply scripts/Entity.gd:25]: Will be done in the next merge

File: scripts/Mob.gd (hunk 3 above)
Comment (uSirThunder) [reply scripts/Mob.gd:12]: Will be done in the next merge

**Pull request merged**